src/edu/csus/ecs/pc2/ui/ResultsComparePane.java

```diff
@@ -0,0 +1,461 @@
+
+        String primaryCCSDirectory = primaryCCSResultsDirectoryTextField.getText();
+
+        if (showErrorMessage("Enter a primary CCS results directory", StringUtilities.isEmpty(primaryCCSDirectory))) {
+            return;
+        }
+
+        if (showErrorMessage("Primary CCS directory does not exist, pick an existing results directory", !directoryExists(primaryCCSDirectory))) {
+            return;
+        }
+
+        if (showErrorMessage("Enter a pc2 results directory", StringUtilities.isEmpty(pc2ResultsDirectory))) {
+            return;
+        }
+        
+        if (showErrorMessage("Missing pc2 directory - use Export Results to create results files", !directoryExists(pc2ResultsDirectory))) {
+            return;
+        }
+
+        try {
+            
+            ResultsCompareReport report = new ResultsCompareReport(getContest(), getController(), primaryCCSDirectory, pc2ResultsDirectory, showDetailsCheckbox.isSelected(), MISSING_SOURCE_TITLE, MISSING_TARGET_TITLE);
+            String[] lines = report.createReport(new Filter());
+            for (String string : lines) {
+                addTextAreaLine(string);
+            }
+
+        } catch (Exception e) {
+            getLog().log(Level.WARNING, "Exception trying export and compare results" + e.getMessage(), e);
+            showMessage(this, "Problem comparing results", "Error writing or comparing results " + e.getMessage());
+        }
+        
+    }
+
+    protected void clearResults() {
+        textArea.setText(null);
+    }
+
+    @Override
+    public void setContestAndController(IInternalContest inContest, IInternalController inController) {
+        super.setContestAndController(inContest, inController);
+
+        SwingUtilities.invokeLater(new Runnable() {
+            public void run() {
+                populateGUI();
+            }
+        });
+    }
+
+    protected void populateGUI() {
+
+        ClientSettings clientSet = getClientSettings(getContest());
+        String exportDir = clientSet.getProperty(ClientSettings.PC2_RESULTS_DIR);
+        pc2ResultsDirectoryTextField.setText(exportDir);
+        exportDirectoryLabel.setToolTipText(exportDir);
+
+        String primaryResultsDir = clientSet.getProperty(ClientSettings.PRIMARY_CCS_RESULTS_DIR);
+        primaryCCSResultsDirectoryTextField.setText(primaryResultsDir);
+        
+        textArea.removeAll();
+        
+        addTextAreaLine("Click "+SHOW_COMPARISON_BUTTON_TITLE+" to show comparison");
+    }
+    
+    private void addTextAreaLine(String string) {
+        textArea.append(string + Constants.NL);
+    }
+
+    private void updateUserSetting(String key, String value) {
+
+        try {
+            updateClientSettings(key, value);
+        } catch (Exception e) {
+            StaticLog.getLog().log(Level.WARNING, "Problem updating settings?", e);
+            e.printStackTrace(); // TODO 760 how to report problem to user ?
```

Comment:
Regarding the question asked in the comment above, what's wrong with adding the line
```
    showErrorMessage("Problem updating client attribute '" + key + "' to value '" + value + "'", true);
```
to the `catch` clause?

Reply:
From what I can see, in response to my comment about adding a call to `ShowErrorMessage()`, you instead replaced the call to `printStackTrace()` with a call to `Utilities.printStackTrace()`.  While that's a better stack trace printout, it doesn't have the same effect as what I proposed -- which was to produce a pop-up dialog letting the user know of the problem.  I don't think putting stack traces on the console is as good a solution as giving the user a dialog to look at...

Reply:
I was working through error handling because improving UI error handling has been a someday task for years.

I have pushed a new change that uses the FrameUtilities showExceptionMessage  method for showing error messages and exceptions to the user.   At this time it is a bare bones method, as part of CI (short for CIP short for Continuous Improvement Process) there will be a new Issue to vastly improve access to error information, by doing things like providing a way to copy the stack trace into the clipboard, provide a way to copy to the clipboard information for bug reporting, etc.  I plan to start using  (once this PR is approved) that method for any/all UI error reporting.

Another improvement will be to make some error messages/report non-modal and add to a list so we can actually report on some exceptions without halting the user's work.